router/xgress_edge_tunnel_v2/hosted.go

```diff
@@ -115,6 +120,21 @@ func (self *hostedServiceRegistry) run() {
 	}
 }
 
+func (self *hostedServiceRegistry) pruneTerminatorIdCache() {
+	cache := self.env.GetRouterDataModel().GetTerminatorIdCache()
+
+	var toRemove []string
+	cache.IterCb(func(key string, v string) {
+		if !self.terminators.Has(v) {
+			toRemove = append(toRemove, key)
+		}
+	})
+
+	for _, key := range toRemove {
```

Comment:
dunno if it would be useful to have a log before/after this operation to have an indication of how long it takes?

Reply:
I would expect this to take milliseconds at most as the number of services should not be large and it's simple data structure operations